src/GUI/RevEng.Core.50/CommentCSharpEntityTypeGenerator.cs

```diff
@@ -398,7 +424,23 @@ protected virtual void GenerateNavigationProperties([NotNull] IEntityType entity
 
                     var referencedTypeName = navigation.TargetEntityType.Name;
                     var navigationType = navigation.IsCollection ? $"ICollection<{referencedTypeName}>" : referencedTypeName;
-                    _sb.AppendLine($"public virtual {navigationType} {navigation.Name} {{ get; set; }}");
+
+                    string nullableAnnotation = string.Empty;
+                    string defaultAnnotation = string.Empty;
+
+                    if (_nullableReferences && !navigation.IsCollection)
+                    {
+                        if (navigation.ForeignKey?.IsRequired == true)
```

Comment:
Is this evaluation correct?
If FK (ForeignKey) is set to "Null" (nullable) in SQL, then I would assume FK.IsRequired will be "false" (not "true")

Maybe it should be "inverted"?
```
                        if (navigation.ForeignKey?.IsRequired == true)
                        {
                            defaultAnnotation = $" = default!;";
                        }
                        else
                        {
                            nullableAnnotation = "?";
                        }
```